pkg/pillar/pubsub/socketdriver/publish.go

```diff
@@ -149,21 +157,23 @@ func (s *Publisher) serveConnection(conn net.Conn, instance int) {
 	sendToPeer := make(pubsub.LocalCollection)
 	sentRestarted := false
 	// Read request
-	buf := make([]byte, 65536)
-	res, err := conn.Read(buf)
+	buf := bufPool.Get().(*[]byte)
+	defer bufPool.Put(buf)
+
+	res, err := conn.Read(*buf)
```

Comment:
This doesn't help much since we block in Read. Thus with N connections from subscribers will will have N goroutines blocked in Read each using a buffer.

If you call os.Select with infinite timeout  before Read, and only allocate a buffer when select says it is readable, then the number of buffers in use should be significantly lower since only those currently processing data will need a buffer.

Reply:
Looks like you don't need to use select but can call
		res, err := s.sock.Read([]byte{})
and once that returns there is data available and you get Get the buffer.

Also, I thought you said defer in the loop is a bad idea; should you explicitly do a Put() at the end of the loop?